### ISSUE TITLE
python 3.7 / Windows 7 / venv / fails to find python library

### ISSUE THREAD

**Issue by terrdavis** [pyinstaller/pyinstaller#4162]
Building from a venv in python 3.7, I get a cryptic error (see out.txt [here](https://gist.github.com/terrdavis/9061aab083e7d435889554624d500a3b)).
I also tried this with PyInstaller version 3.5.dev0+8c27d2668c, with the same error. I tracked down the root cause, which is that `sys.executable` is not with it's `python37.dll` brother.  

Also in the gist:  
* diff1 shows the simplest way I could get it to work.  
* diff2 (changes made to 3.5.dev0+8c27d2668c) shows the changes I made that will:
    * prevent the cryptic error message (`get_python_library_path` is only used in two places, one of which raises an error if nothing is found, so I moved the error to the source)
    * run the fallback only when the default method fails on windows

**Comment by chris360KS** [comment]
Same problem after upgrading Python to 3.7.3 - from the log it looks like you are also using 3.7.3. Looks like they changed somehing around `sys.executable` here : https://bugs.python.org/issue28441

Not sure if this has broken PyInstaller, but might be helpful for tracking this down.

**Comment by terrdavis** [comment]
It worked as expected for me when I wasn’t in a venv. The problem I saw was PyInstaller looking in the directory of sys.executable, which doesn't include python37.dll when in a venv. The solution I found was to make sure PyInstaller looked at the global executable directory as a fallback.

**Comment by chris360KS** [comment]
Ahh thanks for the temporary solution @terrdavis , Will give that a shot.

**Comment by rex-87** [comment]
@terrdavis your temporary solution (diff1) works like a charm, thank you.

**Comment by troyjfarrell** [comment]
Confirming that the diff2 patches provided by @terrdavis fix this problem with PyInstaller 3.4, Python 3.7.2 in a venv on Windows 10.  Thanks!

**Comment by terrdavis** [comment]
Thanks for the feedback @troyjfarrell! I'm glad this helped someone.
@chris360KS is there any chance that [diff2](https://gist.github.com/terrdavis/9061aab083e7d435889554624d500a3b) can be incorporated?

**Comment by chris360KS** [comment]
@terrdavis  I'm not a contributor to this project, I'm just using pyinstaller on one of my own projects - but can confirm another way of making it work was just adding the sys executable dir to the Analysis path. Below is my build.spec file which runs well on PyInstaller 3.4, Python 3.7.3, Windows 10:

```python

# -*- mode: python -*-

block_cipher = None

a = Analysis(['./prod_sendout/main.py'],
             pathex=["./prod_sendout", "./", "C:/Python37"],
             binaries=[],
             datas=[('prod_sendout/*.yaml', '.'),  # Include settings files
                    ('prod_sendout/*.json', '.'),
                    ('data', 'data')],
             hiddenimports=["prod_sendout"],
             hookspath=[],
             runtime_hooks=[],
             excludes=[],
             win_no_prefer_redirects=False,
             win_private_assemblies=False,
             cipher=block_cipher,
             noarchive=False)
pyz = PYZ(a.pure, a.zipped_data,
             cipher=block_cipher)
exe = EXE(pyz,
          a.scripts,
          [],
          exclude_binaries=True,
          name='ProdSendout',
          debug=True,
          bootloader_ignore_signals=False,
          strip=False,
          upx=True,
          console=True)

coll = COLLECT(exe,
               a.binaries,
               a.zipfiles,
               a.datas,
               strip=False,
               upx=True,
               name='prod_sendout')
```

**Comment by terrdavis** [comment]
Ahh, okay @chris360KS. Good to know there is another workaround.  
It looks like I will need to create a pull request myself based on the the sheer quantity of issues and PRs.

**Comment by bjones1** [comment]
This might be fixed by 80941bcdbfdb2baabdaafa023f1e2abbc45b0218 -- reopen if not.

**Comment by terrdavis** [comment]
The issue is fixed!
However I think it's a good idea to move this `IOError`: 
from
https://github.com/pyinstaller/pyinstaller/blob/d7405248998c89e75c8963f267c0ddad333889e4/PyInstaller/building/build_main.py#L681
to https://github.com/pyinstaller/pyinstaller/blob/d7405248998c89e75c8963f267c0ddad333889e4/PyInstaller/depend/bindepend.py#L935 

because `bindepend.get_python_library_path` is also called from: https://github.com/pyinstaller/pyinstaller/blob/d7405248998c89e75c8963f267c0ddad333889e4/PyInstaller/building/api.py#L273
  
This produces a cryptic `TypeError` (that I had to track down manually) instead of the more helpful `IOError` that is specific to `bindepend.get_python_library_path`.
